coderd/files.go

```diff
@@ -86,7 +86,9 @@ func (api *API) fileByHash(rw http.ResponseWriter, r *http.Request) {
 	}
 	file, err := api.Database.GetFileByHash(r.Context(), hash)
 	if errors.Is(err, sql.ErrNoRows) {
-		httpapi.Forbidden(rw)
+		httpapi.Write(rw, http.StatusNotFound, httpapi.Response{
```

Comment:
this reintroduces the security concern that now you can distinguish between whether it exists or not even if you have no access.  We should _also_ return 404 on line 102 of this file.

Reply:
I see what you mean. Before we were returning 403 to be consistent, but that is not a good UX.

If authorize returns a 404 in some cases, can that be a bad UX? 

----

My question is mainly, are these security concerns ones we should take as important right now. If we return a 403 vs 404 we do leak the info so the user knows if the resource exists or not. Is that a security issue we should take seriously right now? (Honest question)

Reply:
So it look like I need to make `Authorize` return a 404 message consistent with the existing 404 message

Reply:
Yeah.  It would be good to have one 404 code path that we use everywhere so we can standardize messaging.

Reply:
@spikecurtis Will fix this now.